cmd/machine-api-migration/main.go

```diff
@@ -153,6 +159,35 @@ func main() {
 		os.Exit(0)
 	}
 
+	configClient, err := configv1client.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		klog.Errorf("failed to create config client: %s", err)
+		os.Exit(1)
+	}
+
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("failed to fetch infrastructure: %s", err)
+		os.Exit(1)
+	}
```

Comment:
You could use `mgr.GetClient()` and use the controller runtime client rather than constructing a typed client if you prefer